include/treelite/tree_impl.h

```diff
@@ -925,6 +925,7 @@ ModelImpl<ThresholdType, LeafOutputType>::InitFromPyBuffer(
 
   auto tree_hanlder = [&begin](Tree<ThresholdType, LeafOutputType>& tree) {
     tree.InitFromPyBuffer(begin, begin + kNumFramePerTree);
+    begin += kNumFramePerTree;
```

Comment:
Since we're capturing `begin` by reference, won't this iterate begin forward as a side effect of executing the lambda? It's not immediately clear to me why that is the correct thing to do with this handler.  Maybe add a comment or refactor such that the lambda returns the next iterator to be processed rather than directly changing `begin`?

Reply:
> Since we're capturing begin by reference, won't this iterate begin forward as a side effect of executing the lambda?

Yes, this is intended behavior.

> refactor such that the lambda returns the next iterator to be processed

That would not be feasible, since the tree handler for the file stream does not involve iterators.

Reply:
Great! Can we get a comment just to make it clear to future developers that the side-effect is intentional?